modules/publicize/publicize.php

```diff
@@ -418,15 +419,102 @@ function save_meta( $post_id, $post ) {
 		// Next up will be ::publicize_post()
 	}
 
+	public function update_published_message( $messages ) {
+		global $post_type, $post_type_object, $post;
+		if ( ! $this->post_type_is_publicizeable( $post_type ) ) {
+			return $messages;
+		}
+		$view_post_link_html = '';
+		$viewable = is_post_type_viewable( $post_type_object );
+		if ( $viewable ) {
+			$view_text = __( 'View post' ); // intentionally omitted domain
+
+			if ( 'jetpack-portfolio' == $post_type ) {
+				$view_text = esc_html__( 'View project', 'jetpack' );
+			}
+
+			$view_post_link_html = sprintf( ' <a href="%1$s">%2$s</a>',
+				esc_url( get_permalink( $post ) ),
+				$view_text
+			);
+		}
+
+		$services = $this->get_publicizing_services( $post->ID );
+		if ( empty( $services ) ) {
+			return $messages;
+		}
+
+		$labels = array();
+		foreach ( $services as $service => $display_names ) {
+			$labels[] = sprintf(
+				/* translators: Service name is %1$s, and account name is %2$s. */
+				_x( '%1$s (%2$s)', 'Service and account string', 'jetpack' ),
+				esc_html( $service ),
+				esc_html( implode( ', ', $display_names ) )
+			);
+		}
+
+		$messages['post'][6] = sprintf(
+			/* translators: Services and account string is %1$s, link to view published post is %2$s */
+			_x( 'Post published and sharing on %1$s. %2$s', 'Post published notice', 'jetpack' ),
+			implode( ', ', $labels ),
+			$view_post_link_html
+		);
+
+		if ( $post_type == 'post' && class_exists('Jetpack_Subscriptions' ) ) {
+			$subscription = Jetpack_Subscriptions::init();
+			if ( $subscription->should_email_post_to_subscribers( $post ) ) {
+				$messages['post'][6] = sprintf(
+					/* translators: Services and account string is %1$s, link to view published post is %2$s */
+               		_x( 'Post published, sending emails to subscribers and sharing post on %1$s. %2$s', 'Post published notice', 'jetpack' ),
+                	implode( ', ', $labels ),
+                	$view_post_link_html
+				);
+			}
+		}
+
+		$messages['jetpack-portfolio'][6] = sprintf(
+			/* translators: Published Message: services and account string is %1$s, link to view published post is %2$s */
+			_x( 'Project published and sharing project on %1$s. %2$s', 'Post published notice', 'jetpack' ),
+			implode( ', ', $labels ),
```

Comment:
Should `,` be translated in this context? I.e. would this work in left-to-right text?

Reply:
@akirk Any suggestions here?

Reply:
That will be ok in RTL.

Reply:
Please remove the `%2$s` here. This is more confusing to translators than helpful. Thanks!

Reply:
(i.e. just append the link outside of the translate function)